### PR TITLE
fix: dmn uploads

### PR DESCRIPTION
**Pull request by mbritense**
Closes https://ritense.tpondemand.com/entity/71945-be-cant-upload-dmn

### CODE REVIEW

File: core/src/main/java/com/ritense/valtimo/service/CamundaProcessService.java [hunk 1]
```diff
@@ -229,14 +234,24 @@ public void deleteAllProcesses(String processDefinitionKey, String reason) {
...
+                throw new ProcessNotUpdatableException("Process is not eligible to be deployed.");
+            }
+
+            repositoryService.createDeployment().addModelInstance(fileName, bpmnModel).deploy();
+        } else if (fileName.endsWith(".dmn")) {
+            DmnModelInstance dmnModel = Dmn.readModelFromStream(fileInput);
+
+            repositoryService.createDeployment().addModelInstance(fileName, dmnModel).deploy();
+        } else {
+            throw new IllegalArgumentException("File to deploy is not valid BPMN or DMN.");
```
Comment (marijnritense) [inline core/src/main/java/com/ritense/valtimo/service/CamundaProcessService.java]: Can you mention the filename in this exception as well?

File: core/src/main/java/com/ritense/valtimo/service/CamundaProcessService.java [hunk 2]
```diff
@@ -229,14 +234,24 @@ public void deleteAllProcesses(String processDefinitionKey, String reason) {
...
-        if (!isDeployable(model)) {
-            throw new ProcessNotUpdatableException("Process is not eligible to be deployed.");
-        }
 
-        repositoryService.createDeployment().addModelInstance(processName, model).deploy();
+        if (fileName.endsWith(".bpmn")) {
+            BpmnModelInstance bpmnModel = Bpmn.readModelFromStream(fileInput);
+
+            if (!isDeployable(bpmnModel)) {
+                throw new ProcessNotUpdatableException("Process is not eligible to be deployed.");
```
Comment (marijnritense) [inline core/src/main/java/com/ritense/valtimo/service/CamundaProcessService.java]: Can you mention the filename in this exception as well?

File: core/src/test/java/com/ritense/valtimo/service/CamundaProcessServiceIntTest.java [hunk 3]
```diff
@@ -63,6 +66,21 @@ void shouldDeployNewProcess() throws IOException, ProcessNotUpdatableException {
         Assertions.assertTrue(definitions.stream().anyMatch(processDefinition -> processDefinition.getKey().equals("deployedProcess")));
     }
 
+    @Test
```
Comment (marijnritense) [inline core/src/test/java/com/ritense/valtimo/service/CamundaProcessServiceIntTest.java:73]: Can you also write a test to verify a non-dmn non-bpmn file is handled correctly?

File: core/src/main/java/com/ritense/valtimo/exception/FileExtensionNotSupportedException.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+package com.ritense.valtimo.exception;
+
+public class FileExtensionNotSupportedException extends Exception {
+
+    /**
+     * Constructs a new exception with the specified detail message.  The
+     * cause is not initialized, and may subsequently be initialized by
+     * a call to {@link #initCause}.
+     *
+     * @param   fileExtension   The name of the file extension that is not supported
```
Comment (Klaas-Ritense) [inline core/src/main/java/com/ritense/valtimo/exception/FileExtensionNotSupportedException.java]: unnecessary comment

File: core/src/main/java/com/ritense/valtimo/exception/NoFileExtensionFoundException.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+package com.ritense.valtimo.exception;
+
+public class NoFileExtensionFoundException extends Exception {
+
+    /**
+     * Constructs a new exception with the specified detail message.  The
+     * cause is not initialized, and may subsequently be initialized by
+     * a call to {@link #initCause}.
+     *
+     * @param   fileName   The name of the file which has no extension
```
Comment (Klaas-Ritense) [inline core/src/main/java/com/ritense/valtimo/exception/NoFileExtensionFoundException.java]: please remove this unnecessary comment

File: core/src/main/java/com/ritense/valtimo/exception/ProcessNotDeployableException.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+package com.ritense.valtimo.exception;
+
+public class ProcessNotDeployableException extends Exception {
+
+    /**
+     * Constructs a new exception with the specified detail message.  The
+     * cause is not initialized, and may subsequently be initialized by
+     * a call to {@link #initCause}.
+     *
+     * @param   fileName   The name of the bpmn file that was to be deployed
```
Comment (Klaas-Ritense) [inline core/src/main/java/com/ritense/valtimo/exception/ProcessNotDeployableException.java]: unnecessary comment

**Review by Klaas-Ritense — APPROVED** [review]

**Pull request merged**